client/utils/planning.ts

```diff
@@ -1223,15 +1224,24 @@ function getCoverageIcon(
     return coverageIcons[type]?.[iconType] ?? iconForUnknownType;
 }
 
-function getCoverageIconColor(coverage: IPlanningCoverageItem): string {
-    if (coverage.workflow_status === COVERAGES.WORKFLOW_STATE.ACTIVE) {
-        return 'var(--sd-colour-success)';
-    } else if (get(coverage, 'assigned_to.state') === ASSIGNMENTS.WORKFLOW_STATE.COMPLETED) {
-        return 'var(--sd-colour-success)';
-    } else if (isCoverageDraft(coverage) || get(coverage, 'workflow_status') === COVERAGES.WORKFLOW_STATE.ACTIVE) {
-        return 'var(--sd-colour-highlight)';
+function getCoverageIconColor(item: IPlanningCoverageItem, field = 'state'): string {
+    if (item.workflow_status === 'cancelled') {
+        return 'var(--sd-colour-state--canceled)';
+    }
+
+    switch (getItemWorkflowState(item.assigned_to, field)) {
+    case ASSIGNMENTS.WORKFLOW_STATE.ASSIGNED:
+        return 'var(--sd-colour-state--in-workflow)';
+    case ASSIGNMENTS.WORKFLOW_STATE.IN_PROGRESS:
+        return 'var(--sd-colour-state--in-progress)';
+    case ASSIGNMENTS.WORKFLOW_STATE.COMPLETED:
+        return 'var(--sd-colour-state--completed)';
+    }
+
+    if (item.assigned_to.user === null || item.assigned_to.desk === null) {
```

Comment:
Avoid using negatives in naming and flow. For example, it wouldn't be a good name `const notUnassigned= true` because it is difficult to understand. In a similar manner, your `if` condition is whether "unassigned" label should be applied. Instead use the condition to determine if `assigned` should be applied and use "else" for unassigned.